### PR TITLE
Align selection field with how existing apps use selection

### PR DESCRIPTION
**Pull request by elfisher**
Fix selection field to use list of options rather than options object.
Add strings to Mira Kit
Fix request proxy parameters to be identical with what `fetch` accepts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ Presentation property definitions are dictionaries that require that you specify
 - `selection`: A series of options.
   - `exclusive`: boolean, optional. Prevents multiple selection.
   - `options`: list, required.
-    - `name`: string, required.
```
Comment (KiaFathi) [inline README.md]: Aren't name and label different? @elfisher

Comment (KiaFathi) [inline README.md]: I think they are both required here.

File: src/components/inspectorFields/SelectionField.js [hunk 2]
```diff
@@ -24,16 +20,12 @@ class SelectionField extends React.Component {
     const presentationProperty = this.props.presentationProperty;
```
Comment (KiaFathi) [inline src/components/inspectorFields/SelectionField.js]:
```
const { presentationProperty } = this.props;
const { name, value } = presentationProperty
```

File: src/helpers.js [hunk 3]
```diff
@@ -48,18 +48,22 @@ function valAppVar(appVar, presProp) {
       const options = presProp.options;
       // if it is exclusive make sure value is a valid selection from pres props
       if (presProp.exclusive) {
-        return (
-          appVar.label in options && appVar.value === options[appVar.label]
-        );
+        return options.some(option => {
+          return appVar.label === option.label && appVar.value === option.value;
```
Comment (KiaFathi) [inline src/helpers.js]: This is not equivalent to ^

File: src/helpers.js [hunk 4]
```diff
@@ -48,18 +48,22 @@ function valAppVar(appVar, presProp) {
...
+        });
       }
       // if not exclusive check each selection is valid
-      return appVar.reduce(
-        (isValid, selection) =>
+      return appVar.reduce((isValid, selection) => {
+        return (
```
Comment (KiaFathi) [inline src/helpers.js]: Don't need a return here.

File: src/helpers.js [hunk 5]
```diff
@@ -48,18 +48,22 @@ function valAppVar(appVar, presProp) {
...
           isValid &&
-          selection.label in options &&
-          selection.value === options[selection.label],
-        true,
-      );
+          options.some(option => {
+            return (
```
Comment (KiaFathi) [inline src/helpers.js]: Don't need a return here.

File: src/components/inspectorFields/SelectionField.js (hunk 2 above)
Comment (sadu99) [inline src/components/inspectorFields/SelectionField.js]: +1

File: src/helpers.js (hunk 3 above)
Comment (elfisher) [inline src/helpers.js]: @KiaFathi `appVar = { label, value}` needs to only be equivalent to 1 of the options in selection for it to be a valid selection.

Comment (elfisher) [inline src/helpers.js]: Is there a better way to validate this?

File: README.md (hunk 1 above)
Comment (elfisher) [inline README.md]: From -> https://github.com/JedWatson/react-select

```
Options should be provided as an Array of Objects, each with a value and label property for rendering and searching. You can use a disabled property to indicate whether the option is disabled or not.
```

Comment (KiaFathi) [inline README.md]: Sure, that's react select. That's not Mira specific though.

Comment (elfisher) [inline README.md]: Is there a reason why we would want to diverge here?

Comment (KiaFathi) [inline README.md]: For our options it appears we need a value and a name:
```
              value={option.value}
              disabled={!!option.disabled}
              children={option.name}```

based on inspector field view.

Comment (KiaFathi) [inline README.md]: Looking at our DB, our apps use name and value, but do not use label.

Comment (KiaFathi) [inline README.md]: See both weather and flyers

File: src/components/AppContainer.js [hunk 6]
```diff
@@ -28,6 +29,7 @@ class AppContainer extends React.Component {
       applicationVariables,
       MiraFileRequestResource,
```
Comment (KiaFathi) [inline src/components/AppContainer.js]: miraFileRequestResource
miraRequestResource

File: README.md [hunk 7]
```diff
@@ -166,6 +166,7 @@ The `miraRequestProxy` prop provides an API for making HTTP and HTTPS requests.
 
 The `miraRequestProxy` has an identical API to the [whatwg-fetch specification](https://github.github.io/fetch/).
```
Comment (KiaFathi) [inline README.md]: update to new name.

update to new name.

**Pull request merged**